data/README.md

```diff
@@ -0,0 +1,108 @@
+system is constantly processing events (clicks, purchases, etc) and displaying
+the latest results in a user interface.
+
+*Parallel Processing*: performing multiple tasks at the same time, for example
+on different cores or processors. Parallel processing is necessary in order to
+perform more than one computation at once; common uses are parsing or aggregation.
+
+*Concurrent Processing*: managing multiple ongoing tasks at once without
+necessarily processing more than one task in the same exact moment. Concurrent
+processing is required in order to perform more than one effect at once, such as
+waiting for multiple network requests to complete.
+
+*Constant Memory*: processing a stream where the amount of memory required does
+not increase with the size of the stream.
+
+*At Least Once Delivery*: a guarantee that a given message will be delivered at
+least once, but may be delivered more than once. This is achieved in Kafka by
+committing an offset after it's been fully processed and in RabbitMQ by
+acknowledging a message after fully processing it.
+
+*At Most Once Delivery*: a guarantee that a message will never be delivered more
+than once, but may not be delivered at all. This is achieved in Kafka by
+committing an offset before fully processing a message and in RabbitMQ by
+acknowledging a message before fully processing it.
+
+*Distributed Data Processing*: breaking up data into partitions so that large
+amounts of data can be processed by many machines simultaneously.
+
+*Cluster*: several computers (or virtual machines) grouped together to perform a
+single task.
+
+*Scala*: a programming language (like Ruby, Python, or JavaScript) which is fast
+and has become popular for data-focused tasks. Scala runs on the Java Virtual
+Machine, which is a high-performance engine for running languages like Scala
+that compile into bytecode.
+
+*Type Safety*: languages that provide type safety (such as Scala) check the
+program for possible errors as part of the compilation process, which allows developers to prevent
+many types of bugs before being deployed.
+
+*Spark*: a distributed computing engine for big data and data streams. Spark is
+a Scala-focused framework for data engineering and data science.
+
+*Kafka*: a distributed commit log for data streams. Many of the large data
+systems deployed today use Kafka.
+
+*Record Stream*: a stream where each message is an independent, unique record
+which does not replace a previous record in the stream.
+
+*Changelog Stream*: a stream where each message represents the latest state for
+a particular entity.
+
+*Topic*: in Kafka, a partitioned, append-only log of messages which can be
+consumed in order by partition.
+
+*Partition*: in Kafka, a way of breaking the messages of a topic into groups
+which can be consumed in parallel by one or more workers.
+
+*Queue*: in RabbitMQ, messages sent to an exchange are placed on a queue.
+Messages on a queue can be consumed in parallel by one or more workers.
```

Comment:
A few more terms for consideration to add:

```
*Consumer*: An application or process that reads from a data stream.

*Producer*: An application or process that writes to a data stream.
```

Reply:
Thanks! Added in 1d12c71.